core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -621,6 +621,14 @@ default void setFieldType(ConnectorSession session, ConnectorTableHandle tableHa
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support setting field types");
     }
 
+    /**
+     * Set a not null constraint to the specified column
```

Comment:
The Javadoc should clarify that the engine does not check the nullability of existing data, and that this is the responsibility of the connectors.